File: docs/snippets/csharp/docker/usage.cs
```csharp title="usage.cs"
using System;
using System.Diagnostics;
using System.IO;
using System.Text.Json;
using System.Threading.Tasks;

var dockerClient = new DockerXbergLib();

try
{
    await dockerClient.StartContainerAsync();
    await Task.Delay(2000);

    var content = await dockerClient.ExtractAsync("document.pdf");
    Console.WriteLine($"Extracted content:\n{content}");
}
finally
{
    await dockerClient.StopContainerAsync();
}

class DockerXbergLib
{
    private const string ContainerName = "xberg-api";
    private const string ContainerImage = "xberg:latest";
    private const int ApiPort = 8000;

    public async Task StartContainerAsync()
    {
        Console.WriteLine("Starting Xberg Docker container...");

        var processInfo = new ProcessStartInfo
        {
            FileName = "docker",
            Arguments = $"run -d --name {ContainerName} -p {ApiPort}:8000 {ContainerImage}",
            UseShellExecute = false,
            RedirectStandardOutput = true,
        };

        using (var process = Process.Start(processInfo))
        {
            await process.WaitForExitAsync();
        }

        Console.WriteLine($"Container started on http://localhost:{ApiPort}");
    }

    public async Task<string> ExtractAsync(string filePath)
    {
        using (var client = new HttpClient())
        {
            var fileBytes = await File.ReadAllBytesAsync(filePath);
            using (var content = new MultipartFormDataContent())
            {
                content.Add(new ByteArrayContent(fileBytes), "file", Path.GetFileName(filePath));

                var response = await client.PostAsync(
                    $"http://localhost:{ApiPort}/api/extract",
                    content
                );

                response.EnsureSuccessStatusCode();
                var json = await response.Content.ReadAsStringAsync();
                var result = JsonSerializer.Deserialize<JsonElement>(json);
                return result.GetProperty("content").GetString();
            }
        }
    }

    public async Task StopContainerAsync()
    {
        Console.WriteLine("Stopping Xberg Docker container...");

        var processInfo = new ProcessStartInfo
        {
            FileName = "docker",
            Arguments = $"stop {ContainerName}",
            UseShellExecute = false,
        };

        using (var process = Process.Start(processInfo))
        {
            await process.WaitForExitAsync();
        }

        processInfo.Arguments = $"rm {ContainerName}";
        using (var process = Process.Start(processInfo))
        {
            await process.WaitForExitAsync();
        }

        Console.WriteLine("Container stopped and removed");
    }
}
```

File: docs/snippets/java/docker/Usage.java
```java title="Usage.java"
import java.io.*;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.UUID;
import com.google.gson.JsonParser;

public final class Usage {
  private static final String BOUNDARY = "----WebKitFormBoundary" + UUID.randomUUID();
  private final String containerName;
  private final int apiPort;

  public Usage(String containerName, int apiPort) {
    this.containerName = containerName;
    this.apiPort = apiPort;
  }

  public void startContainer(String image) throws IOException, InterruptedException {
    System.out.println("Starting Xberg Docker container...");
    ProcessBuilder pb = new ProcessBuilder("docker", "run", "-d",
      "--name", containerName,
      "-p", apiPort + ":8000",
      image);
    Process process = pb.start();
    if (process.waitFor() != 0) {
      throw new RuntimeException("Failed to start container");
    }
    System.out.println("Container started on http://localhost:" + apiPort);
  }

  public String extract(String filePath) throws IOException {
    byte[] fileBytes = Files.readAllBytes(Paths.get(filePath));
    String fileName = Paths.get(filePath).getFileName().toString();

    URL url = new URL("http://localhost:" + apiPort + "/api/extract");
    HttpURLConnection conn = (HttpURLConnection) url.openConnection();
    conn.setRequestMethod("POST");
    conn.setRequestProperty("Content-Type", "multipart/form-data; boundary=" + BOUNDARY);
    conn.setDoOutput(true);

    try (OutputStream os = conn.getOutputStream()) {
      os.write(("--" + BOUNDARY + "\r\n").getBytes());
      os.write(("Content-Disposition: form-data; name=\"file\"; filename=\"" + fileName + "\"\r\n").getBytes());
      os.write("Content-Type: application/octet-stream\r\n\r\n".getBytes());
      os.write(fileBytes);
      os.write(("\r\n--" + BOUNDARY + "--\r\n").getBytes());
    }

    StringBuilder response = new StringBuilder();
    try (BufferedReader reader = new BufferedReader(new InputStreamReader(conn.getInputStream()))) {
      String line;
      while ((line = reader.readLine()) != null) {
        response.append(line);
      }
    }

    return JsonParser.parseString(response.toString())
      .getAsJsonObject()
      .get("content")
      .getAsString();
  }

  public void stopContainer() throws IOException, InterruptedException {
    System.out.println("Stopping Xberg Docker container...");
    new ProcessBuilder("docker", "stop", containerName).start().waitFor();
    new ProcessBuilder("docker", "rm", containerName).start().waitFor();
    System.out.println("Container stopped and removed");
  }

  public static void main(String[] args) throws Exception {
    Usage docker = new Usage("xberg-api", 8000);

    try {
      docker.startContainer("xberg:latest");
      Thread.sleep(2000);

      String content = docker.extract("document.pdf");
      System.out.println("Extracted content:\n" + content);
    } finally {
      docker.stopContainer();
    }
  }
}
```
